buffer/src/main/java/io/netty/buffer/AdaptivePoolingAllocator.java

```diff
@@ -584,13 +611,16 @@ boolean trySetNextInLine(Chunk chunk) {
 
         void free() {
             // Release the current Chunk and the next that was stored for later usage.
+            Chunk next = NEXT_IN_LINE.getAndSet(this, MAGAZINE_FREED);
+            if (next != null && next != MAGAZINE_FREED) {
+                next.release();
+            }
```

Comment:
Just get rid of the duplicated code ?
```suggestion
            restoreMagazineFreed();
```